src/ray/core_worker/context.cc

```diff
@@ -184,7 +184,7 @@ void WorkerContext::SetCurrentTask(const TaskSpecification &task_spec) {
   RAY_CHECK(current_job_id_ == task_spec.JobId());
   if (task_spec.IsNormalTask()) {
     current_task_is_direct_call_ = true;
-    runtime_env_ = task_spec.RuntimeEnv();
+    serialized_runtime_env_ = task_spec.SerializedRuntimeEnv();
```

Comment:
This should never change because the workers are cached based on the runtime env. Can we just set it once and then add a `RAY_CHECK` that we never see a new one?

Reply:
That makes a lot of sense!  Let me make a note to do that once worker caching is in place, unless you think it's better to do it in this PR.

Reply:
It's probably ok to do in a follow-up